### ISSUE TITLE
CustomResources created outside of the Ingress Controller are deleted

### ISSUE THREAD

**Issue by jonstacks** [ngrok/kubernetes-ingress-controller#267]
### What happened

I manually created a `tunnel.ingress.k8s.ngrok.com` resource with custom labels that I wanted running:

```yaml
kind: Tunnel
apiVersion: ingress.k8s.ngrok.com/v1alpha1
metadata:
  name: starwars-tunnel
  namespace: default
spec:
  forwardsTo: starwars.default.svc.cluster.local:23
  labels:
    k8s.ngrok.com/namespace: default
    k8s.ngrok.com/service: starwars
    k8s.ngrok.com/port: "23"
```

After a while, the tunnel was no longer present. During the driver's sync, it was calculated as being able to be deleted [here](https://github.com/ngrok/kubernetes-ingress-controller/blob/main/internal/store/driver.go#L285-L288)

### What you think should happen instead

The Tunnel should not have been deleted by the Ingress Controller. Since it wasn't created from an Ingress and was manually created, it should have remained intact until someone deleted it.

### How to reproduce

1. Create a `tunnel.ingress.k8s.ngrok.com` manually like the one specified above. 
2. Verify the Tunnel exists: `kubectl get tunnels.ingress.k8s.ngrok.com` . You should see the tunnel you just created.
3. Trigger a sync through one of the following methods and wait a few minutes
  a. Edit the annotations of an ingress
  b. Do a `rollout restart` of the ingress controller deployment.
4. Verify the Tunnel is gone: `kubectl get tunnels.ingress.k8s.ngrok.com`

**Comment by CK-Ward** [comment]
cc @pshafton

**Comment by nikolay-ngrok** [comment]
I've gone ahead and implemented most of this using annotations in #282 . However, I realized that our migration logic is somewhat flowed - if you are running a bunch of custom tunnels without an ingress, nothing will be marked. The controller will assume its not migrated and will mark everything as controller owned, and then immediately delete it all as part of the `Sync`. I'm looking for/exploring other options on how we can detect if we need to run the migration or not.
